lib/storage/environment.go

```diff
@@ -107,9 +109,41 @@ func (r *EnvironmentV1) GetKeyValues() map[string]string {
 
 // CheckAndSetDefaults validates this resource and sets defaults
 func (r *EnvironmentV1) CheckAndSetDefaults() error {
+	var errors []error
+	if err := r.checkProxy(constants.HTTPProxyEnvVar); err != nil {
```

Comment:
One other Nitpick I just thought of (sorry). The golang code and I think other implementations accept both lower and upper case versions of the environment variable. 

https://github.com/golang/go/blob/master/src/vendor/golang.org/x/net/http/httpproxy/proxy.go#L93-L94